Tests/HtcMock/Server/src/SampleComputerService.cs

```diff
@@ -189,25 +163,69 @@ public override async Task<Output> Process(ITaskHandler taskHandler)
                          .ConfigureAwait(false);
       }
 
-      output = new Output
+      var taskError = taskHandler.TaskOptions.GetValueOrDefault("TaskError",
+                                                                string.Empty);
+
+      if (taskError != string.Empty && taskHandler.TaskId.EndsWith(taskError))
+      {
+        logger_.LogInformation("Return Deterministic Error Output");
+        return new Output
                {
-                 Ok = new Empty(),
+                 Error = new Output.Types.Error
+                         {
+                           Details = "Deterministic Error",
+                         },
                };
+      }
+
+      var taskRpcException = taskHandler.TaskOptions.GetValueOrDefault("TaskRpcException",
+                                                                       string.Empty);
+
+      if (taskRpcException != string.Empty && taskHandler.TaskId.EndsWith(taskRpcException))
+      {
+        throw new RpcException(new Status(StatusCode.Internal,
+                                          "Deterministic Exception"));
+      }
+
+      return new Output
+             {
+               Ok = new Empty(),
+             };
     }
-    catch (Exception ex)
+    catch (RpcException ex)
     {
+      var taskRpcException = taskHandler.TaskOptions.GetValueOrDefault("TaskRpcException",
+                                                                       string.Empty);
+      if (taskRpcException != string.Empty && taskHandler.TaskId.EndsWith(taskRpcException))
+      {
+        Thread.Sleep(TimeSpan.FromSeconds(15));
```

Comment:
What is the purpose of this timeout?

Reply:
It was a test. It is not necessary. I will remove it.